sklearn/utils/_array_api.py

```diff
@@ -669,6 +669,25 @@ def _average(a, axis=None, weights=None, normalize=True, xp=None):
     return sum_ / scale
 
 
+def _median(x, axis=None, keepdims=False, xp=None):
+    # `median` is not included in the Array API spec, but is implemented in most
+    # array libraries, and all that we support (as of May 2025).
```

Comment:
```suggestion
    # XXX: `median` is not included in the array API spec, but is implemented
    # in most array libraries, and all that we support (as of May 2025).
    # TODO: simplify this code to use scipy instead once the oldest supported
    # SciPy version provides `scipy.stats.quantile` with native array API
    # support (likely scipy 1.6 at the time of writing).
```

Reply:
Interesting.
I did think about what we should do in the future. Ralf suggested maybe `quantile` should go into array api extra (I need to open a RFC issue about this in array api extra).

Despite most array libraries having a `median` (including dask), which may be somewhat faster than `scipy.stats.quantile` (benchmarking required), would you still be inclined to use `scipy.stats.quantile` over `median` of the native library?
(I don't think `median` will be in the spec, mostly because `quantile` is more versatile and the torch's version of `median` implementation differs from everyone else)